### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by ikhoon**
- Armeria 1.23.1 -> 1.24.0
- Curator 5.3.0 -> 5.5.0
- Jackson 2.14.2 -> 2.15.2
- JGit 6.5.0 -> 6.6.0
- Micrometer 1.10.5 -> 1.11.1
- sshd 2.9.2 -> 2.10.0
- Spring Boot 2.7.10 -> 2.7.12, 3.0.5 -> 3.1.0
- Zookeeper 3.5.8 -> 3.7.1
- Build
  - docker-remote-api 6.7.0 -> 9.3.1
  - Guava 31.1 -> 32.0.1
  - jmh-gradle-plugin 0.7.0 -> 0.7.1
  - json-unit 2.37.0 -> 2.38.0
  - JUnit 5.9.2 -> 5.9.3
  - node-gradle-plugin 3.5.1 -> 5.0.0

git subrepo pull --force gradle/scripts

subrepo:
  subdir:   "gradle/scripts"
  merged:   "9edc7a86"
upstream:
  origin:   "https://github.com/line/gradle-scripts.git"
  branch:   "main"
  commit:   "9edc7a86"
git-subrepo:
  version:  "0.4.5"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "dbb99be"

### CODE REVIEW

**Review by jrhee17 — APPROVED** [review]
Looks good once the CI is fixed to pass

File: dependencies.toml [hunk 1]
```diff
@@ -3,27 +3,68 @@
...
+thrift09 = { strictly = "0.9.3-1" }
+zookeeper = "3.7.1"
 
 [boms]
-armeria = { module = "com.linecorp.armeria:armeria-bom", version = "1.23.1" }
-jackson = { module = "com.fasterxml.jackson:jackson-bom", version = "2.14.2" }
-junit5 = { module = "org.junit:junit-bom", version = "5.9.2" }
+armeria = { module = "com.linecorp.armeria:armeria-bom", version.ref = "armeria" }
+jackson = { module = "com.fasterxml.jackson:jackson-bom", version.ref = "jackson" }
+junit5 = { module = "org.junit:junit-bom", version.ref = "junit5" }
```
Comment (ikhoon) [inline dependencies.toml:63]: Sorry for the messing up. 😅

**Comment by ikhoon** [comment]
All failures have been fixed, expect `yarn` task failed to run on Windows CI. It may not be a big problem since we build tarball on Linux and `yarn` will be removed when the new UI is released.

**Review by minwoox — APPROVED** [review]
Thanks!!!

**Comment by ikhoon** [comment]
Thanks for the quick review. 🙇‍♂️

**Pull request merged**